jetty-security/src/main/java/org/eclipse/jetty/security/ConstraintSecurityHandler.java

```diff
@@ -260,12 +262,19 @@ public static Constraint createConstraint(String name, String[] rolesAllowed, Em
     }
 
     /**
-     * @return Returns the constraintMappings.
+     * @return only the durable mappings if we are not yet started, otherwise
+     * return both the durables and transients.
      */
     @Override
     public List<ConstraintMapping> getConstraintMappings()
     {
-        return _constraintMappings;
+        //if we've started, then we've processed both the durable and
+        //transient constraint mappings
+        if (isRunning())
+            return _constraintMappings;
+        else
+            //otherwise, we only have durables
+            return _durableConstraintMappings;
```

Comment:
Is this necessary?  The `_durableConstraintMappings` is a subset of `_constraintMappings`, so isn't it always OK to return `_constraintMappings`?

Reply:
Oh I see.... will comment more in request for change....